images.CI/linux-and-win/convert-to-vhd.ps1

```diff
@@ -0,0 +1,211 @@
+  }
+
+  $targetBlobUri = 'https://{0}.blob.core.windows.net/{1}/{2}?{3}' -f $StorageAccountName, $ContainerName, $ImageBlobName, $targetSasToken
+  return $targetBlobUri
+}
+
+$ErrorActionPreference = "Stop"
+
+# Check if azcopy is installed
+if (-not $env:AZCOPYPATH) {
+  Write-Error "AZCOPYPATH is not set"
+  exit 1
+}
+Write-Host "azcopy path is set"
+
+# Login to Azure
+az login --service-principal --username $ClientId --password $ClientSecret --tenant $TenantId | Out-Null
+
+# Create Compute Image Gallery if it doesn't exist
+Write-Host "Creating Compute Image Gallery '$GalleryName'..."
+$galleryExists = az sig list --resource-group $ResourceGroupName --query "[?name=='$GalleryName']" -o tsv
+if ($null -eq $galleryExists) {
+  az sig create --resource-group $ResourceGroupName --gallery-name $GalleryName --location $Location
+}
+
+# Create Image Definition if it doesn't exist
+Write-Host "Creating Image Definition '$ManagedImageName'..."
+$imageDefinitionName = "RunnerImage-$GalleryImageSku"
+if ($ManagedImageName -like "*windows*") {
+  $imageOsType = "Windows"
+} elseif ($ManagedImageName -like "*ubuntu*") {
+  $imageOsType = "Linux"
+} else {
+  throw "Unknown OS type for image '$ManagedImageName'"
+}
+
+$galleryImageExists = az sig image-definition list --resource-group $ResourceGroupName --gallery-name $GalleryName --query "[?name=='$imageDefinitionName']" -o tsv
+if ($null -eq $galleryImageExists) {
+  az sig image-definition create `
+    --resource-group $ResourceGroupName `
+    --gallery-name $GalleryName `
+    --gallery-image-definition $imageDefinitionName `
+    --publisher "GuthubImagesGeneration" `
+    --offer "RunnerImage" `
+    --sku $GalleryImageSku `
+    --os-type $imageOsType `
+    --location $Location
+}
+
+# Create Image Version from existing Managed Image
+Write-Host "Creating Image Version '$GalleryImageVersion' from Managed Image '$ManagedImageName'..."
+az sig image-version create `
+  --resource-group $ResourceGroupName `
+  --gallery-name $GalleryName `
+  --gallery-image-definition $imageDefinitionName `
+  --gallery-image-version $GalleryImageVersion `
+  --managed-image "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName/providers/Microsoft.Compute/images/$ManagedImageName" `
+  --target-regions $Location `
+  --replica-count 1 `
+  --location $Location
+
+# Create Azure Managed Disk from Shared Image Gallery
+Write-Host "Creating Azure Managed Disk '$ManagedImageName' from Shared Image Gallery..."
+az disk create `
+  --resource-group $ResourceGroupName `
+  --name $ManagedImageName `
+  --location $Location `
+  --gallery-image-reference "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName/providers/Microsoft.Compute/galleries/$GalleryName/images/$imageDefinitionName/versions/$GalleryImageVersion"
+
+# Generate SAS URL for the Managed Disk
+Write-Host "Generating SAS URL for the Managed Disk '$ManagedImageName'..."
```

Comment:
same as above, hmm, but not sure we can get rid of sas here exactly

Reply:
It looks like SAS is the only option here.